xcube_server/controllers/tiles.py

```diff
@@ -117,6 +124,39 @@ def array_image_id_factory(level):
     return tile
 
 
+def get_legend(ctx: ServiceContext,
+               ds_name: str,
+               var_name: str,
+               params: RequestParams):
+    cmap_cbar = params.get_query_argument('cbar', default=None)
+    cmap_vmin = params.get_query_argument_float('vmin', default=None)
+    cmap_vmax = params.get_query_argument_float('vmax', default=None)
+    if cmap_cbar is None or cmap_vmin is None or cmap_vmax is None:
+        default_cmap_cbar, default_cmap_vmin, default_cmap_vmax = ctx.get_color_mapping(ds_name, var_name)
+        cmap_cbar = cmap_cbar or default_cmap_cbar
+        cmap_vmin = cmap_vmin or default_cmap_vmin
+        cmap_vmax = cmap_vmax or default_cmap_vmax
+
+    try:
+        cmap = cm.get_cmap(cmap_cbar)
+    except ValueError:
+        raise ServiceResourceNotFoundError(f"color bar {cmap_cbar} not found")
+
+    fig = plt.figure(figsize=(1, 5))
+    ax1 = fig.add_subplot(1, 1, 1)
+    norm = matplotlib.colors.Normalize(vmin=cmap_vmin, vmax=cmap_vmax)
+    image_legend = matplotlib.colorbar.ColorbarBase(ax1, cmap=cmap,
+                                                    norm=norm, orientation='vertical')
+    image_legend.set_label(ctx.get_legend_lable(ds_name, var_name))
```

Comment:
Note that `ctx.get_legend_lable()` can return `None`. Does `image_legend.set_label()` accept that?